src/go/k8s/Dockerfile

```diff
@@ -3,22 +3,29 @@ FROM --platform=$BUILDPLATFORM public.ecr.aws/docker/library/golang:1.21.3 as bu
 ARG TARGETARCH
 ARG TARGETOS
 
+RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
```

Comment:
This makes me pretty uneasy... Generally we'd want to have a set of static keys in the repository or pre-populated on the build machines. This method technically makes us vulnerable to a supply chain attack as someone could masquerade as `GitHub.com` and feed us fake keys.

IIRC, Github's API provides a static list of their SSH keys. We should be able to drop those into a file and copy it in or even paste them into this Dockerfile and `cat` them with a HEREDOC.

Reply:
the change was made to allow us to build locally. Without these changes I was not able to run "docker-build" on my machine, but the build process does work on the pipeline I can revert this, which is why its on a different commit.